File: seed/csharp-model/trace/src/SeedTraceClient/Problem/ProblemFiles.cs
using System.Text.Json.Serialization;
using SeedTraceClient;

namespace SeedTraceClient;

public class ProblemFiles
{
    [JsonPropertyName("solutionFile")]
    public FileInfo SolutionFile { get; init; }

    [JsonPropertyName("readOnlyFiles")]
    public List<FileInfo> ReadOnlyFiles { get; init; }
}

File: seed/rust-model/trace/src/problem_files.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ProblemFiles {
    #[serde(rename = "solutionFile")]
    pub solution_file: FileInfo,
    #[serde(rename = "readOnlyFiles")]
    pub read_only_files: Vec<FileInfo>,
}